beamer/tests/contracts/test_request_manager.py

```diff
@@ -296,6 +296,31 @@ def test_claim_nonexistent_request(request_manager):
         request_manager.claimRequest(1234, FILL_ID, {"from": claimer})
 
 
+def test_claim_request_extension(request_manager, token, claim_stake):
+    """
+    Test that claiming is allowed around expiry
+    and will revert after validUntil + claimRequestExtension
+    """
+    (requester,) = alloc_accounts(1)
+    (claimer,) = alloc_whitelisted_accounts(1, {request_manager})
+    token.mint(requester, 1, {"from": requester})
+    request_id = make_request(request_manager, token, requester, requester, 1)
+
+    valid_until = request_manager.requests(request_id)[RM_R_FIELD_VALID_UNTIL]
+    claim_request_extension = request_manager.claimRequestExtension()
+    # test that request expiration does not prevent claiming
+    timestamps = [valid_until - 1, valid_until, valid_until + claim_request_extension - 1]
+
+    for timestamp in timestamps:
+        chain.sleep(timestamp - chain.time())
```

Comment:
I like the fact we are testing all the "interesting" timestamps.
Is there a way to ensure that `sleep` did the trick, e.g. something like
```
assert chain.time() == timestamp
```
so that we also ensure this does not silently break in the future?

Reply:
Unfortunately I don't think so and with new versions of either brownie or ganache, this can break again.
The reason for this whole mess was a major change in ganache v6 to v7 where the next transaction mined would increment the timestamp. In v7 this is not the case anymore and you need to explicitly increment the timestamp for a new block.

Brownie then, decided to increment the timestamp internally before setting the time in ganache upon `chain.mine()`. They did this to be consistent with v6. So now, if one calls `chain.mine(timestamp=x)`, brownie internally increments the timestamp before calling ganache which is super counter intuitive to me. 

At least they mention in there docs that for timestamp testing one should use chain.sleep. I personally like the fact that I can control the timestamp for the next block in ganache, especially for this kind of testing.